File: apps/todo_service/src/app/Domain/Services/TodoService.php
<?php

namespace App\Domain\Services;

use App\Domain\DTOs\TodoDTO;
use App\Domain\Models\Todo;
use App\Domain\Repositories\TodoRepositoryInterface;

class TodoService
{
    protected TodoRepositoryInterface $todoRepository;

    function __construct(TodoRepositoryInterface $todoRepository)
    {
        $this->todoRepository = $todoRepository;
    }

    function getAll()
    {
        return $this->todoRepository->getAll();
    }

    function get(int $id): Todo
    {
        return $this->todoRepository->get($id);
    }

    function create(TodoDTO $todoDTO): Todo
    {
        return $this->todoRepository->create($todoDTO);
    }

    function update(int $id, TodoDTO $todoDTO): Todo
    {
        return $this->todoRepository->update($id, $todoDTO);
    }

    function destroy(int $id)
    {
        return $this->todoRepository->destroy($id);
    }
}

File: apps/python_service/app/domain/services/todo_service.py
from app.domain.repositories.todo_repository import TodoRepository
from app.domain.dtos.todo_dto import TodoDTO


class TodoService:
    
    todo_repository: TodoRepository

    def __init__(self, todo_repository: TodoRepository):
        self.todo_repository = todo_repository

    def get_all(self):
        return self.todo_repository.get_all()

    def get(self, id: int):
        return self.todo_repository.get(id)

    def create(self, todo_dto: TodoDTO):
        return self.todo_repository.create(todo_dto)

    def update(self, id: int, todo_dto: TodoDTO):
        return self.todo_repository.update(id, todo_dto)

    def destroy(self, id: int):
        return self.todo_repository.destroy(id)

